test/addons/dlopen-ping-pong/test.js

```diff
@@ -14,7 +14,5 @@ process.dlopen(module, bindingPath,
 module.exports.load(`${path.dirname(bindingPath)}/ping.so`);
 assert.strictEqual(module.exports.ping(), 'pong');
 
-// Check that after the addon is loaded with
-// process.dlopen() a require() call fails.
-const re = /^Error: Module did not self-register\.$/;
-assert.throws(() => require(`./build/${common.buildType}/binding`), re);
+// This second `require()` call should not throw an error.
+require(`./build/${common.buildType}/binding`);
```

Comment:
This is a pretty strong change in behaviour. `NODE_MODULE` and even `NODE_MODULE_CONTEXT_AWARE` were so far expected to fail to register a second time around. Developers should explicitly move to modules that can be loaded multiple times by replacing those two macros with `NODE_MODULE_INIT(/*exports, module, context*/) {...}` as per [our docs](https://nodejs.org/docs/latest/api/addons.html#addons_context_aware_addons), and ensuring that they address the concerns raised there.

If we require that module move to `NODE_MODULE_INIT` then that's another reason we don't need to store `dlopen` handles in a file-name-keyed table.